### DISCUSSION TITLE
Native DirectX/Vulkan Draw/compute calls

### DISCUSSION THREAD

**Discussion by kariem2k** [DiligentGraphics/DiligentEngine#286] [General]
Hello,

I'm curious if it's possible to issue direct DirectX/Vulkan draw and compute calls from a legacy project while incorporating Diligent Engine for new portions of code. My goal is to eventually transition the entire project to use Diligent Engine, but initially, I'd like to utilize both APIs. Does Diligent Engine offer any functions that reset the graphics state, allowing me to safely execute native DirectX code? These calls would be done in separate render passes.

Thank you.

**Comment by kariem2k** [comment]
I apologize I just found this 

https://github.com/DiligentGraphics/DiligentCore?tab=readme-ov-file#low_level_api_interoperability

**Comment by TheMostDiligent** [comment]
> Does Diligent Engine offer any functions that reset the graphics state

Use the [InvalidateState](https://github.com/DiligentGraphics/DiligentCore/blob/1b0e29bf0e44486e207cff97b90f8409f43f7611/Graphics/GraphicsEngine/interface/DeviceContext.h#L2518) method to reset internal state tracking.
